mobile/src/reducers/last_online_reducer.js

```diff
@@ -0,0 +1,24 @@
+import { SET_LAST_ONLINE_TIME } from '../actions/types';
+
+const initialState = {
+  lastOnlineTime: undefined,
+};
+
+export default function (state = initialState, action) {
+  switch (action.type) {
+    case SET_LAST_ONLINE_TIME: {
+      const isConnected = action.payload;
+      const newState = Object.assign({}, state);
```

Comment:
We use object destructuring approach for this. In your case:
return { ...state, lastOnlineTime: isConnected ? null : new Date().getTime()  }

Reply:
Yeah makes sense, I took the solution from some other reducer that's already there, but I love how concise destructuring is.